### ISSUE TITLE
Not able to create SSHClient when building release

### ISSUE THREAD

**Issue by MrHazee** [shaqian/flutter_ssh#37]
Hi,

First I had the problem with no internet connection as mentioned here #32 but adding the flag --no-shrink solved that problem. But I still cant connect nor create a SSHClient. Everything works as expected on iOS and in Android debug mode. I am having a hard time to debug this as well.  But I´ve narrowed it down to where it fails: 
"var client = new SSHClient(....)" 
Can I catch any exception when new SSHClient fails?
Do I need to add (except from INTERNET) any android permissions to use SSH?
Ive tried searching but no success.

EDIT: 
It fails when I run `result = await client.connect();`

Im also getting this warning (as well in debug mode):
"Note: /Users/razbook3.0/development/flutter/.pub-cache/hosted/pub.dartlang.org/ssh-0.0.6/android/src/main/java/sq/flutter/ssh/SshPlugin.java uses unchecked or unsafe operations."
Thanks /Rasmus

**Comment by MrHazee** [comment]
Actually, I got it to work. I tried the "android/app/proguard-rules.pro" mod as suggested here [https://github.com/shaqian/flutter_ssh/issues/27#issuecomment-599180850](url) and it worked for me!

**Comment by tc14077** [comment]
How can you possible to narrowed the bug down? I am facing same problem too, my application runs fine in debug mode but fail after release build.

**Comment by tc14077** [comment]
I found a way to identify this bug, and the situation is also like you mentioned early. My application is running normally under the debug mode, but It fails when I run `result = await client.connect()` under the release build. Unfortunately, I am not able to fix it as suggested in #27 .

Update:
The error message is `java.lang.ClassNotFoundException: com.jcraft.jsch.jce.Random`. It is captured by following code. I also tested with 1.12.13 and 1.17.5 flutter sdk version.
```
try {
      result = await targetClient.connect();
      makeHTTPGet(
          "my.testing.server.com/client_connect_success");
      if (result == "session_connected") {
        result = await targetClient.connectSFTP();
        if (result == "sftp_connected") {
          result = "connected";
        }
      }
    } on PlatformException catch (e) {
      //Error handling for SSH/SFTP connect fail.
      print('Error: ${e.code}\nError Message: ${e.message}');
      makeHTTPGet(
          "my.testing.server.com/client_connect_fail${e.message}");
      result = "error";
    }
```

**Comment by tc14077** [comment]
Update:
It seems that I had solved issues. The reason that missing the `com.jcraft.jsch.jce.Random` class is the class was stripped out by the building tools. Therefore, this class will 'exist' and work fine when the developers are using debug mode but not in release build. 

To solve this issue, I performed following steps:

1. Updating my Android Gradle Plugin Version to **3.5.0** which was **3.3.1** in the past. Meantime, the Gradle Version should also being update to **5.4.1**. The Gradle Version can be updated by changing the **distributionUrl** in **android/gradle/wrapper/gradle-wrapper.properties**.

`distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip`

<img width="898" alt="Screenshot 2020-08-13 at 11 39 41 AM" src="https://user-images.githubusercontent.com/59038619/90091736-c4b14080-dd59-11ea-9c6a-41fbb5937651.png">


2. As [#27(comment)](https://github.com/shaqian/flutter_ssh/issues/27#issuecomment-599180850) mention, developers are required to edit **android/app/build.gradle** file. **Enabling minifyEnabled** and **making a reference** to the `proguard-rules.pro` file. Please notice that, I have not included the **useProguard** option in this case because android no longer uses ProGuard to perform compile-time code optimization on the Android Gradle plugin 3.4.0 or higher.  For more details, please see [android documentation](https://developer.android.com/studio/build/shrink-code#keep-code).

```
buildTypes {
        release {
            // TODO: Add your own signing config for the release build.
            // Signing with the debug keys for now, so `flutter run --release` works.
            signingConfig signingConfigs.debug
            
            // Enables code shrinking, obfuscation, and optimization for only
            // your project's release build type.
            minifyEnabled true

            // Enables resource shrinking, which is performed by the
            // Android Gradle plugin.
            shrinkResources true

            // Includes the default ProGuard rules files that are packaged with
            // the Android Gradle plugin. To learn more, go to the section about
            // R8 configuration files.
            proguardFiles getDefaultProguardFile(
                    'proguard-android-optimize.txt'),
                    'proguard-rules.pro'
        }
    }
```

3. Open a `proguard-rules.pro` file under the **android/app/** directory with following contents. With the **printconfiguration** option, the build tool will output a full report of all the rules that R8 applies when building your project, include the following in your module’s **proguard-rules.pro** file. 
```
#Flutter Wrapper
-printconfiguration /any/directory/you/want/full-r8-config.txt
-keep class io.flutter.app.** { *; }
-keep class io.flutter.plugin.**  { *; }
-keep class io.flutter.util.**  { *; }
-keep class io.flutter.view.**  { *; }
-keep class io.flutter.**  { *; }
-keep class io.flutter.plugins.**  { *; }
-keep class com.jcraft.**  { *; }
```
